src/core/algorithms/dd/dd_verifier/dd_verifier.cpp

```diff
@@ -0,0 +1,211 @@
+    std::byte const *first_value = column.GetValue(tuple_pair.first);
+    std::byte const *second_value = column.GetValue(tuple_pair.second);
+    auto const &type = static_cast<model::IMetrizableType const &>(column.GetType());
+    dif = type.Dist(first_value, second_value);
+    return dif;
+}
+
+std::vector<std::pair<int, int>> DDVerifier::GetRowsWhereLhsHolds() const {
+    std::vector<std::pair<int, int>> result;
+    std::vector<model::ColumnIndex> columns;
+    for (auto const &constraint : dd_.left) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    auto curr_constraint = dd_.left.cbegin();
+    for (std::size_t i = 0; i < num_rows_; i++) {
+        for (std::size_t j = i + 1; j < num_rows_; j++) {
+            if (auto const dif = CalculateDistance(columns[0], {i, j});
+                curr_constraint->constraint.Contains(dif)) {
+                result.emplace_back(i, j);
+            }
+        }
+    }
+    ++curr_constraint;
+    for (std::size_t i = 1; i < columns.size(); i++) {
+        std::vector<std::pair<int, int>> new_result;
+        for (std::size_t j = 0; j < result.size(); j++) {
+            if (double const dif = CalculateDistance(columns[i], result[j]);
+                curr_constraint->constraint.Contains(dif)) {
+                new_result.emplace_back(result[j]);
+            }
+        }
+        result = std::move(new_result);
+        ++curr_constraint;
+    }
+    return result;
+}
+
+void DDVerifier::LoadDataInternal() {
+    typed_relation_ = model::ColumnLayoutTypedRelationData::CreateFrom(*input_table_, false);
+}
+
+void DDVerifier::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kDDString});
+}
+
+void DDVerifier::CheckCorrectnessDd() const {
+    for (auto const &constraint : dd_.left) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+    for (auto const &constraint : dd_.right) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+}
+
+unsigned long long DDVerifier::ExecuteInternal() {
+    std::vector<model::ColumnIndex> columns;
+    for (auto const &constraint : dd_.left) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    for (auto const &constraint : dd_.right) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
```

Comment:
Nit: You could consider
```suggestion
    std::vector<model::ColumnIndex> columns;
    columns.reserve(dd_.left.size() + dd_.right.size());

    auto add_columns = [this, &columns](const auto& constraints) {
        std::ranges::transform(constraints, std::back_inserter(columns), [this](const auto& c) {
            return typed_relation_->GetSchema()->GetColumn(c.column_name)->GetIndex();
        });
    };

    add_columns(dd_.left);
    add_columns(dd_.right);
```